compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1339,7 +1339,16 @@ class RefChecks extends MiniPhase { thisPhase =>
   }
 
   override def transformTypeTree(tree: TypeTree)(using Context): TypeTree = {
-    checkExperimental(tree.symbol, tree.srcPos)
+    val tpe = tree.tpe
+    tpe.dealias.foreachPart {
```

Comment:
Let's consider a situation like this:
```scala
object LibApi:
  @deprecated
  trait OldName
  
  type NewName = OldName 
```
when the author of a library decided to rename a trait so they created an nondeprecated type alias pointing to it. Should we raise a warning in this case as well?
Probably it would be better for the author to have
```scala
object LibApi:
  @deprecated
  type OldName = NewName
  
  trait NewName
```
instead but can/should we enforce such a solution?

Reply:
Good example. We should not dealias here.